debug_toolbar/apps.py

```diff
@@ -1,16 +1,58 @@
 from __future__ import absolute_import, unicode_literals
 
+import warnings
+
 from django.apps import AppConfig
+from django.conf import settings
+from django.middleware.gzip import GZipMiddleware
+from django.utils.module_loading import import_string
 from django.utils.translation import ugettext_lazy as _
 
-from debug_toolbar import settings as dt_settings
+from debug_toolbar.middleware import DebugToolbarMiddleware
 
 
 class DebugToolbarConfig(AppConfig):
     name = 'debug_toolbar'
     verbose_name = _("Debug Toolbar")
 
     def ready(self):
-        if dt_settings.get_patch_settings():
-            dt_settings.patch_all()
-        dt_settings.check_middleware()
+        check_middleware()
+
+
+def check_middleware():
```

Comment:
If not too complicated, that might be a candidate for using the system check framework.